inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -650,15 +651,26 @@ void RemovePermutationsNHWCToNCHWPass::run() {
         if (getInputTo(l->outData.front()).empty()) {
             continue;
         }
+
         auto next = getInputTo(l->outData.front()).begin()->second;
-        auto prev = CNNNetPrevLayer(l);
+        while (!LayerInfo(next).isPermute() && !LayerInfo(next).isNonFunctional() && !LayerInfo(next).isInput() &&
+               next->outData.size() == 1) {
+            auto input_to = getInputTo(next->outData.front());
+            if (input_to.size() != 1) break;
+            next = input_to.begin()->second;
+        }
 
         // The next layer must be NCHW to NHWC permute
         if (!LayerInfo(next).isPermute() || next->input()->getLayout() != Layout::NCHW ||
             next->GetParamAsInts("order") != GetPermuteOrder(Layout::NCHW, Layout::NHWC)) {
             continue;
         }
 
+        auto parent = CNNNetPrevLayer(l);
+        auto prev = parent;
+        while (!LayerInfo(prev).isPermute() && !LayerInfo(prev).isNonFunctional() && !LayerInfo(next).isInput()) {
+            prev = CNNNetPrevLayer(prev);
```

Comment:
what if there is no previous layer?

Reply:
I've added CNNNetHasPrevLayer() checks for l and prev.